tensorflow/compiler/xla/service/hlo_computation.cc

```diff
@@ -919,7 +937,11 @@ Status HloComputation::ReplaceInstruction(HloInstruction* old_instruction,
       ShapeUtil::Compatible(old_instruction->shape(), new_instruction->shape()))
       << ShapeUtil::HumanString(old_instruction->shape()) << " vs "
       << ShapeUtil::HumanString(new_instruction->shape());
+  return ReplaceInstructionWithDifferentShape(old_instruction, new_instruction);
+}
 
+Status HloComputation::ReplaceInstructionWithDifferentShape(
+    HloInstruction* old_instruction, HloInstruction* new_instruction) {
```

Comment:
In line 974, it should call ReplaceAllUsesWithDifferentShape instead of ReplaceAllUsesWith
We already know that we can enter this method with different shapes of old_instruction and new_instruction.